src/test/scala/com/atomist/rug/runtime/js/EditorTest.scala

```diff
@@ -442,33 +497,64 @@ class EditorTest extends FlatSpec with Matchers {
     assert(p.getMaxLength === 100)
   }
 
-  private  def invokeAndVerifyConstructed(tsf: FileArtifact): (JavaScriptProjectEditor, SuccessfulModification) = {
+  it should "load empty repo in generator" in {
+    val tsf = SimpleFileBasedArtifactSource(
+      StringFileArtifact(".atomist/generators/Thing.ts", SimpleGeneratorLoadingEmptyRepo),
+      StringFileArtifact("I dont", "want this"))
+    val as = TypeScriptBuilder.compileWithModel(tsf)
+
+    val jsed = RugArchiveReader(as).generators.head.asInstanceOf[JavaScriptProjectGenerator]
+
+    val prj = jsed.generate("woot", SimpleParameterValues(Map("content" -> "Anders Hjelsberg is God")))
+    jsed
+  }
+
+  it should "load external repo in generator" in {
+    val tsf = SimpleFileBasedArtifactSource(
+      StringFileArtifact(".atomist/generators/Thing.ts", SimpleGeneratorLoadingExternalRepo),
+      StringFileArtifact("I dont", "want this"))
+    val as = TypeScriptBuilder.compileWithModel(tsf)
+
+    val jsed = RugArchiveReader(as).generators.head.asInstanceOf[JavaScriptProjectGenerator]
+
+    val project = SimpleFileBasedArtifactSource(
+      StringFileArtifact("1","1"),
+      StringFileArtifact("2", "2")
+    )
+    val ctx = new ProjectContext(LocalRugContext) {
+      override def repoResolver: Option[RepoResolver] = Some(FixedBranchRepoResolver("atomist", "rug", "master", project))
+    }
+    val prj = jsed.generate("woot", SimpleParameterValues(Map("content" -> "Anders Hjelsberg is God")), ctx)
+    jsed
```

Comment:
Again, I think it would be good to verify that the contents of the generated project have files "1" and "2" and not "I dont".